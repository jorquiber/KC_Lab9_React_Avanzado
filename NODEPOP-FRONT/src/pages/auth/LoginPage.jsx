import { useState } from "react";
import Button from "../../components/shared/Button";
import FormField from "../../components/shared/FormField";
import "./LoginPage.css";
import { useDispatch, useSelector } from "react-redux";
import { authLogin, uiResetError } from "../../store/actions";
import { getUi } from "../../store/selectors";

export default function LoginPage() {
  const dispatch = useDispatch();
  const { pending: isFetching, error } = useSelector(getUi);

  const [formValues, setFormValues] = useState({
    email: "",
    password: "",
  });

  const [isChecked, setIsChecked] = useState(true);

  const handleInputChange = (event) => {
    setFormValues((currentFormValues) => ({
      ...currentFormValues,
      [event.target.name]: event.target.value,
    }));
  };

  const handleCheckboxChange = () => {
    setIsChecked((currentIsChecked) => !currentIsChecked);
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    dispatch(authLogin(formValues, isChecked));
  };

  const resetError = () => dispatch(uiResetError());

  const { email, password } = formValues;
  const buttonDisabled = !email || !password;
  return (
    <div className="loginPage">
      <h1 className="loginPage-title">Log in to Nodepop</h1>
      <form onSubmit={handleSubmit}>
        <FormField
          type="text"
          name="email"
          label="email"
          className="loginForm-field"
          value={email}
          onChange={handleInputChange}
        />
        <FormField
          type="password"
          name="password"
          label="password"
          className="loginForm-field"
          value={password}
          onChange={handleInputChange}
        />
        <Button
          type="submit"
          $variant="primary"
          disabled={buttonDisabled}
          className="loginForm-submit"
        >
          Log in
        </Button>
        <FormField
          type="checkbox"
          name="remember"
          label="Remember me if you restart the browser"
          className="loginForm-field"
          checked={isChecked}
          onChange={handleCheckboxChange}
        />
      </form>
      {error && (
        <div className="loginPage-error" onClick={resetError}>
          {error.message}
        </div>
      )}
    </div>
  );
}
