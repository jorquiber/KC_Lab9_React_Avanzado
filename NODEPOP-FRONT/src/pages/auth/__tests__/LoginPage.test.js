import React from 'react';
import { render } from '@testing-library/react';
import LoginPage from '../LoginPage';
import { Provider } from 'react-redux';

describe('LoginPage', () => {
  const state = { ui: { pending: false, error: null } };
  const store = {
    dispatch: () => {},
    getState: () => state,
    subscribe: () => {},
  };

  const renderComponent = () =>
    render(
      <Provider store={store}>
        <LoginPage />
      </Provider>,
    );

  test('snapshot', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
