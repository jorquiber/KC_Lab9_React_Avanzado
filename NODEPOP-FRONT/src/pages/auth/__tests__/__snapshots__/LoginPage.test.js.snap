// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage snapshot 1`] = `
<div>
  <div
    class="loginPage"
  >
    <h1
      class="loginPage-title"
    >
      Log in to Nodepop
    </h1>
    <form>
      <div
        class="formField loginForm-field"
      >
        <label
          class="formField-label"
        >
          <span>
            email
          </span>
          <input
            autocomplete="off"
            class="formField-input"
            name="email"
            type="text"
            value=""
          />
        </label>
      </div>
      <div
        class="formField loginForm-field"
      >
        <label
          class="formField-label"
        >
          <span>
            password
          </span>
          <input
            autocomplete="off"
            class="formField-input"
            name="password"
            type="password"
            value=""
          />
        </label>
      </div>
      <button
        class="button button-primary loginForm-submit"
        disabled=""
      >
        Log in
      </button>
      <div
        class="formField loginForm-field"
      >
        <label
          class="formField-label"
        >
          <span>
            Remember password if you restart the browser
          </span>
          <input
            autocomplete="off"
            checked=""
            class="formField-input"
            name="remember"
            type="checkbox"
          />
        </label>
      </div>
    </form>
  </div>
</div>
`;
